src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1810,6 +1810,77 @@ internal MethodSymbol EnsureThrowIfNullFunctionExists(SyntaxNode syntaxNode, Syn
+        /// </summary>
+        internal MethodSymbol EnsureThrowSwitchExpressionExceptionParameterlessExists(SyntaxNode syntaxNode, SyntheticBoundNodeFactory factory, DiagnosticBag? diagnostics)
+        {
+            var privateImplClass = GetPrivateImplClass(syntaxNode, diagnostics);
+            var throwSwitchExpressionAdapter = privateImplClass.GetMethod(PrivateImplementationDetails.SynthesizedThrowSwitchExpressionExceptionParameterlessFunctionName);
+            if (throwSwitchExpressionAdapter is not null)
+            {
+                return (MethodSymbol)throwSwitchExpressionAdapter.GetInternalSymbol()!;
+            }
+
+            TypeSymbol returnType = factory.SpecialType(SpecialType.System_Void);
+            var sourceModule = SourceModule;
+
+            // use add-then-get pattern to ensure the symbol exists, and then ensure we use the single "canonical" instance added by whichever thread won the race.
+            privateImplClass.TryAddSynthesizedMethod(new SynthesizedParameterlessThrowMethod(
+                sourceModule,
+                privateImplClass,
+                returnType,
+                PrivateImplementationDetails.SynthesizedThrowSwitchExpressionExceptionParameterlessFunctionName,
+                factory.WellKnownMethod(WellKnownMember.System_Runtime_CompilerServices_SwitchExpressionException__ctor)).GetCciAdapter());
+            return (MethodSymbol)privateImplClass.GetMethod(PrivateImplementationDetails.SynthesizedThrowSwitchExpressionExceptionParameterlessFunctionName)!.GetInternalSymbol()!;
+        }
+
+        /// <summary>
+        /// Creates the ThrowInvalidOperationException helper if needed.
+        /// </summary>
+        internal MethodSymbol EnsureThrowInvalidOperationExceptionExists(SyntaxNode syntaxNode, SyntheticBoundNodeFactory factory, DiagnosticBag? diagnostics)
+        {
+            var privateImplClass = GetPrivateImplClass(syntaxNode, diagnostics);
+            var throwAdapter = privateImplClass.GetMethod(PrivateImplementationDetails.SynthesizedThrowInvalidOperationExceptionFunctionName);
+            if (throwAdapter is not null)
+            {
+                return (MethodSymbol)throwAdapter.GetInternalSymbol()!;
+            }
+
+            TypeSymbol returnType = factory.SpecialType(SpecialType.System_Void);
+            var sourceModule = SourceModule;
+
+            // use add-then-get pattern to ensure the symbol exists, and then ensure we use the single "canonical" instance added by whichever thread won the race.
+            privateImplClass.TryAddSynthesizedMethod(new SynthesizedParameterlessThrowMethod(
+                sourceModule,
+                privateImplClass,
+                returnType,
+                PrivateImplementationDetails.SynthesizedThrowInvalidOperationExceptionFunctionName,
+                factory.WellKnownMethod(WellKnownMember.System_InvalidOperationException__ctor)).GetCciAdapter());
```

Comment:
>factory.WellKnownMethod(WellKnownMember.System_InvalidOperationException__ctor) [](http://example.com/codeflow?start=16&length=79)

Do we have tests where either the exception type is missing or the constructor we need is missing?
You can look for tests that reference those well-known types/members.
This can be tested with MakeTypeMissing/MakeMemberMissing test helpers.
Same question applies for the other two exceptions affected by PR.

Reply:
@jcouv 

- `UnmatchedInput_04` tests `SwitchExpressionException(object)` constructor.
- `UnmatchedInput_02` makes `SwitchExpressionException(object)` missing (or actually it's already missing, the test just didn't declare it explicitly). So this tests the parameterless overload of the constructor.
- `UnmatchedInput_01` doesn't provide `SwitchExpressionException` at all. So this tests `InvalidOperationException`.

There doesn't seem to be any tests that makes `InvalidOperationException` missing though.

None of the tests use `VerifyEmitDiagnostics`, so I'll add it to the existing tests.

I'll:

- Update the existing tests to verify specific ctors are missing.
- Use VerifyEmitDiagnostics in those tests.
- Add new test where InvalidOperationException is also missing.